logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -135,8 +150,15 @@ public void stop() throws Exception {
         LOG.warn("Exception in closing server socket.", e);
       }
     }
-
-    mThreadPool.shutdown();
+    mThreadPool.shutdownNow();
+    // Close all client sockets so that their serving threads can stop. We shut down the threadpool
+    // before closing the sockets so that no new socket threads will be created after we try to
+    // close them all.
+    synchronized (mClientSockets) {
+      for (Socket socket : mClientSockets) {
+        socket.close();
+      }
+    }
     boolean ret = mThreadPool.awaitTermination(THREAD_KEEP_ALIVE_TIME_MS, TimeUnit.MILLISECONDS);
```

Comment:
This seems ineffective given the added call to shutdownNow() on line 153

Reply:
the javadoc for `shutdownNow` says to use `awaitTermination`